boa_engine/src/object/shape/slot.rs

```diff
@@ -0,0 +1,86 @@
+use bitflags::bitflags;
+pub(crate) type SlotIndex = u32;
+
+bitflags! {
+    /// Attributes of a slot.
+    #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    pub struct SlotAttribute: u8 {
+        const WRITABLE = 0b0000_0001;
+        const ENUMERABLE = 0b0000_0010;
+        const CONFIGURABLE = 0b0000_0100;
+        const HAS_GET = 0b0000_1000;
+        const HAS_SET = 0b0001_0000;
```

Comment:
I wonder if these should be `GET` `SET` instead of the `HAS` prefix, to be consistent with the ones above and it makes more sense when looking at:

```rs
    attributes.set(SlotAttribute::GET, self.get().is_some());
    attributes.set(SlotAttribute::SET, self.set().is_some());
```